task.sql

```diff
@@ -3,9 +1,25 @@
+USE ShopDB;
 
--- Some data should be created outside the transaction (here)
+INSERT INTO Products (ID, Name, Description, Price, WarehouseAmount)
+VALUES (1, 'AwersomeProduct', 'An awesome product!', 100, 50);
 
--- Start the transaction 
-START TRANSACTION; 
+INSERT INTO Customers (ID, FirstName, LastName, Email, Address)
+VALUES (1, 'John', 'Doe', 'johndoe@example.com', '123 Main St');
 
--- And some data should be created inside the transaction 
+START TRANSACTION;
 
-COMMIT; 
+INSERT INTO Orders (CustomerID, Date)
+VALUES (1, '2023-01-01');
+
+INSERT INTO OrderItems (OrderID, ProductID, Count)
+VALUES (
+    (SELECT ID FROM Orders WHERE CustomerID = 1 ORDER BY ID DESC LIMIT 1),
```

Comment:
The subquery to get the `OrderID` assumes that the last order inserted is the correct one. Ensure that this logic is correct for your use case, as concurrent transactions might affect the result.